### ISSUE TITLE
Sharp not working on Centos 6.6

### ISSUE THREAD

**Issue by vforv** [lovell/sharp#743]
I tried to install sharp module but not working on Centos 6. Could you help me with this here is my log file:

```
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:305:16)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Linux 3.10.23-xxxx-std-ipv6-64
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /heydoc/node_modules/sharp
gyp ERR! node -v v7.2.1
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
npm verb lifecycle sharp@0.17.2~install: unsafe-perm in lifecycle false
npm verb lifecycle sharp@0.17.2~install: PATH: /usr/local/lib/node_modules/npm/bin/node-gyp-bin:/heydoc/node_modules/sharp/node_modules/.bin:/heydoc/node_modules/.bin:/usr/local/jdk/bin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/opt/cpanel/composer/bin:/usr/local/bin:/usr/X11R6/bin:/root/bin
npm verb lifecycle sharp@0.17.2~install: CWD: /heydoc/node_modules/sharp
npm info lifecycle sharp@0.17.2~install: Failed to exec install script
npm verb unlock done using /root/.npm/_locks/staging-fbe21d76e0022f85.lock for /heydoc/node_modules/.staging
npm verb stack Error: sharp@0.17.2 install: `node-gyp rebuild`
npm verb stack Exit status 1
npm verb stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:255:16)
npm verb stack     at emitTwo (events.js:106:13)
npm verb stack     at EventEmitter.emit (events.js:191:7)
npm verb stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:40:14)
npm verb stack     at emitTwo (events.js:106:13)
npm verb stack     at ChildProcess.emit (events.js:191:7)
npm verb stack     at maybeClose (internal/child_process.js:885:16)
npm verb stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
npm verb pkgid sharp@0.17.2
npm verb cwd /heydoc
npm ERR! Linux 3.10.23-xxxx-std-ipv6-64
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "i" "sharp" "--verbose"
npm ERR! node v7.2.1
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! sharp@0.17.2 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sharp@0.17.2 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the sharp package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs sharp
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls sharp
npm ERR! There is likely additional logging output above.
npm verb exit [ 1, true ]

npm ERR! Please include the following file with any support request:
npm ERR!     /heydoc/npm-debug.log

```

**Comment by lovell** [comment]
Hello, Centos 6 requires libvips to be installed globally. See the "...older Linux-based operating systems such as Centos 6..." bit at http://sharp.dimens.io/en/stable/install/#linux for help.

**Comment by vforv** [comment]
I installed vips:

```
root@dev [/heydoc]# vips --version
vips-7.40.11-Sun Mar 26 17:25:22 Local time zone must be set--see zic manual page 2017
```

**Comment by lovell** [comment]
sharp v0.17.2 requires libvips v8.4.2+ - see http://sharp.dimens.io/en/stable/changelog/#v017-quill

If you're able to use Centos 7 then that will "just work" without any need to install libvips.

**Comment by vforv** [comment]
Unfortunately I am not able to change server... Is there any script to instal 8.4.2 libvips?

**Comment by lovell** [comment]
Did you see http://sharp.dimens.io/en/stable/install/#linux
> For 32-bit Intel CPUs and older Linux-based operating systems such as Centos 6, it is recommended to install a system-wide installation of libvips from source:
> https://github.com/jcupitt/libvips#building-libvips-from-a-source-tarball

**Comment by lovell** [comment]
Closing due to inactivity, please re-open if there are still problems installing libvips from source.

**Comment by trongthanh** [comment]
I would like to reopen this issue. I'm currently installing sharp on a **CentOS 6.8** VPS and am facing a blocker.

What I have done:

1. Install `devtoolset-2` and enable it to get latest gcc 4.8.2
2. Compile libvips from source (I've tried both vips-8.7.0 & 8.6.3 tarball) and install it globally.
I have tried install most of its dependencies and the last of configure output is like below:
```
* build options
native win32:				no
native OS X:				no
open files in binary mode: 		no
enable debug:				no
build deprecated components:		yes
build docs with gtkdoc:			no
gobject introspection: 			no
build vips7 Python binding:		no (default)
install vips8 Python overrides:		no (default)
  (requires pygobject-3.13.0 or later)
build radiance support: 		yes
build analyze support: 			yes
build PPM support: 			yes

* optional dependencies
use fftw3 for FFT: 			no
Magick package: 			none
Magick API version: 			none
load with libMagick: 			no
save with libMagick: 			no
accelerate loops with orc: 		no
  (requires orc-0.4.11 or later)
ICC profile support with lcms: 		no
file import with OpenEXR: 		no
file import with OpenSlide:		no
  (requires openslide-3.3.0 or later)
file import with matio: 		no
PDF import with poppler-glib: 		no
  (requires poppler-glib 0.16.0 or later)
SVG import with librsvg-2.0: 		no
  (requires librsvg-2.0 2.34.0 or later)
zlib: 					yes
file import with cfitsio: 		no
file import/export with libwebp:	no
  (requires libwebp-0.1.3 or later)
support webp metadata:
  (requires libwebpmux-0.5 or later)
text rendering with pangoft2: 		no
file import/export with libpng: 	yes (found by search)
  (requires libpng-1.2.9 or later)
file import/export with libtiff:	no
file import/export with giflib:		no
file import/export with libjpeg:	yes
image pyramid export:			no
  (requires libgsf-1 1.14.26 or later)
use libexif to load/save JPEG metadata: no
```
3. Install node 8.11.1, npm 6.0.0

4. When I run `npm install` on a project that has **sharp** as dependency, I receive the following error:
```
npm i

> sharp@0.20.1 install /var/www/nms-website/node_modules/sharp
> (node install/libvips && node install/dll-copy && prebuild-install) || (node-gyp rebuild && node install/dll-copy)

info sharp Detected globally-installed libvips v8.6.3
info sharp Building from source via node-gyp
make: Entering directory `/var/www/nms-website/node_modules/sharp/build'
  TOUCH Release/obj.target/libvips-cpp.stamp
  CXX(target) Release/obj.target/sharp/src/common.o
In file included from /usr/lib/glib-2.0/include/glibconfig.h:9:0,
                 from /usr/include/glib-2.0/glib/gtypes.h:34,
                 from /usr/include/glib-2.0/glib/galloca.h:34,
                 from /usr/include/glib-2.0/glib.h:32,
                 from /usr/include/glib-2.0/gobject/gbinding.h:30,
                 from /usr/include/glib-2.0/glib-object.h:25,
                 from /usr/local/include/vips/vips8:35,
                 from ../src/common.cc:25:
/usr/include/glib-2.0/glib/gthread.h: In function ‘gboolean g_once_init_enter(volatile gsize*)’:
/usr/include/glib-2.0/glib/gthread.h:347:27: error: size of array is negative
   if G_LIKELY ((gpointer) g_atomic_pointer_get (value_location) != NULL)
                           ^
make: *** [Release/obj.target/sharp/src/common.o] Error 1
make: Leaving directory `/var/www/nms-website/node_modules/sharp/build'
```

I'm totally stuck at this step right now. Cannot get sharp working on CentOS. :(

**Comment by lovell** [comment]
@trongthanh https://stackoverflow.com/questions/10597422/gthread-h-size-of-array-is-negative might help, but I strongly recommend you upgrade to CentOS 7 if possible. Good luck!
